tests/Middleware/AuthTokenMiddlewareTest.php

```diff
@@ -306,6 +285,63 @@ public function testShouldNotifyTokenCallback(callable $tokenCallback)
         $this->assertTrue(MiddlewareCallback::$called);
     }
 
+    public function testAddAuthHeadersFromUpdateMetadata()
+    {
+        $authResult = [
+            'authorization' => 'Bearer 1/abcdef1234567890',
+        ];
+
+        $this->mockFetcher->willImplement(UpdateMetadataInterface::class);
+        $this->mockFetcher->updateMetadata(Argument::cetera())
+            ->shouldBeCalledTimes(1)
+            ->willReturn($authResult);
+        $this->mockFetcher->getLastReceivedToken()
+            ->willReturn(['access_token' => '1/abcdef1234567890']);
+
+        $request = new Request('GET', 'http://foo.com');
+
+        $middleware = new AuthTokenMiddleware($this->mockFetcher->reveal());
+        $mock = new MockHandler([new Response(200)]);
+        $callable = $middleware($mock);
+        $callable($request, ['auth' => 'google_auth']);
+    }
+
+    public function testOverlappingAddAuthHeadersFromUpdateMetadata()
+    {
+        $authHeaders = [
+            'authorization' => 'Bearer 1/abcdef1234567890',
+            'x-goog-api-client' => 'extra-value'
+        ];
+
+        $request = new Request('GET', 'http://foo.com');
+        $request->withHeader('x-goog-api-client', 'default-value');
```

Comment:
This line does nothing, since `withHeader` returns a _new_ Request object, and does not modify the existing request object. It needs to be:
```suggestion
        $request = $request->withHeader('x-goog-api-client', 'default-value');
```

But since it wasn't working before it must not be testing this value as expected.

Reply:
Yeah that line was placed with the purpose of testing how headers are merge but later I figured that testing this functionality is unnecessary. We're not supposed to test how the headers are merged as it's the responsibility of `updateMetadata` method which is out of scope for our context.

Just forgot to remove that line. It's removed now.